### PR TITLE
[DO NOT MERGE] Np 1764/issues with forgot password

### PR DESCRIPTION
**Pull request by olhabn**
@kgrgreer @jlhughes @blakegreen32 @Nauna please let me know what you think regarding comments below

ticket: https://nanopay.atlassian.net/browse/NP-1764

the thing is that there is no user when we resetting password so i have a couple of questions regarding that. 
an alternative is "Success password reset" view

### CODE REVIEW

File: src/foam/nanos/controller/ApplicationController.js [hunk 1]
```diff
@@ -364,7 +364,7 @@ foam.CLASS({
 
       var userNotificationQueryId = this.subject && this.subject.realUser ?
       this.subject.realUser.id : this.user.id;
-
+      //is it called?
```
Comment (olhabn) [inline src/foam/nanos/controller/ApplicationController.js:384]: is` src/foam/nanos/controller/ApplicationController.js initE` method or `nanopay/src/net/nanopay/ui/Controller.js onUserAgentAndGroupLoaded` method 
called for not authorized user?

File: src/foam/nanos/controller/ApplicationController.js [hunk 2]
```diff
@@ -575,6 +575,22 @@ foam.CLASS({
     },
 
     function notify(toastMessage, toastSubMessage, severity, transient) {
+      var userNotificationQueryId = this.subject && this.subject.realUser ?
```
Comment (olhabn) [inline src/foam/nanos/controller/ApplicationController.js:578]: it obviously shouldn't be there but not sure where it should be

File: src/foam/nanos/notification/Notification.js [hunk 3]
```diff
@@ -11,7 +11,7 @@ foam.CLASS({
   implements: [
     'foam.nanos.auth.Authorizable',
     'foam.nanos.auth.CreatedAware',
-    'foam.nanos.auth.CreatedByAware'
+    // 'foam.nanos.auth.CreatedByAware'
   ],
```
Comment (olhabn) [inline src/foam/nanos/notification/Notification.js:15]: what if there is no user?

File: src/foam/nanos/notification/services [hunk 4]
```diff
@@ -28,11 +28,12 @@ p({
...
   "name": "notificationDAO",
   "lazy": false,
+  "authenticate": false,
   "serve": true,
   "serviceScript": """
     return new foam.dao.EasyDAO.Builder(x)
       .setOf(foam.nanos.notification.Notification.getOwnClassInfo())
-      .setAuthorize(true)
+      .setAuthorize(false)
       .setInnerDAO((foam.dao.DAO) x.get("localNotificationDAO"))
```
Comment (olhabn) [inline src/foam/nanos/notification/services:37]: what if there is no user?

File: src/foam/nanos/controller/ApplicationController.js [hunk 5]
```diff
@@ -329,6 +329,23 @@ foam.CLASS({
       this.clientPromise.then(async function(client) {
         self.setPrivate_('__subContext__', client.__subContext__);
```
Comment (olhabn) [inline src/foam/nanos/controller/ApplicationController.js:331]: can we add subscription here?

**Comment by jlhughes** [comment]
Resolved elsewhere.

**Pull request closed without merging**